core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2473,21 +2473,25 @@ public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFile
     }
     // If the file is persisted in UFS, also update corresponding owner/group/permission.
     if ((ownerGroupChanged || permissionChanged) && inode.isPersisted()) {
-      MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
-      String ufsUri = resolution.getUri().toString();
-      UnderFileSystem ufs = resolution.getUfs();
-      if (ownerGroupChanged) {
-        try {
-          ufs.setOwner(ufsUri, inode.getOwner(), inode.getGroup());
-        } catch (IOException e) {
-          throw new AccessControlException("Could not setOwner for UFS file " + ufsUri, e);
+      if ((inode instanceof InodeFile) && !((InodeFile) inode).isCompleted()) {
+        LOG.debug("Alluxio does not propagate chown/chgrp/chmod to UFS for incomplete files.");
```

Comment:
Does it make sense to modify the permission of incomplete files even in Alluxio space?